public/js/maho-captcha.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Maho
+ *
+ * @package     Maho_Captcha
+ * @copyright   Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+const MahoCaptcha = {
+    altchaWidget: null,
+    altchaState: null,
+    frontendSelectors: '',
+    onVerifiedCallback: null,
+
+    async setup(config) {
+        this.altchaWidget = document.querySelector('altcha-widget');
+        this.frontendSelectors = config.frontendSelectors ?? '';
+
+        // Load Altcha JS
+        if (typeof customElements.get('altcha-widget') !== 'function') {
+            await Promise.all([
+                new Promise((resolve, reject) => {
+                    const script = document.createElement('script');
+                    script.src = '/js/altcha.min.js';
+                    script.type = 'module';
+                    script.onload = resolve;
+                    script.onerror = () => reject(`${script.src} Not Found`);
+                    document.head.appendChild(script);
+                }),
+            ]);
+        }
```

Comment:
this loads the altcha script automatically which is something that we have to avoid. my implementation lazy loaded it only when interacting with the form